### ISSUE TITLE
Mistral Small 24B Instruct 2501 GGUF: "shape mismatch on dim 3, 160 <> 128"

### ISSUE THREAD

**Issue by Remember20240719** [EricLBuehler/mistral.rs#1118]
## Describe the bug
1. Download the Q4_K_M GGUF from https://huggingface.co/bartowski/Mistral-Small-24B-Instruct-2501-GGUF/blob/main/Mistral-Small-24B-Instruct-2501-Q4_K_M.gguf
2. Run

```sh
./target/release/mistralrs-server -i gguf --quantized-model-id /download/location --quantized-filename "Mistral-Small-24B-Instruct-2501-Q4_K_M.gguf"
```

Output:
```
INFO mistralrs_server: avx: true, neon: false, simd128: false, f16c: true
INFO mistralrs_server: Sampling method: penalties -> temperature -> topk -> topp -> minp -> multinomial
INFO mistralrs_server: Model kind is: gguf quantized from gguf (no adapters)
INFO mistralrs_core::pipeline::paths: Loading `Mistral-Small-24B-Instruct-2501-Q4_K_M.gguf` locally at `/download/location/Mistral-Small-24B-Instruct-2501-Q4_K_M.gguf`
INFO mistralrs_core::pipeline::gguf: Prompt chunk size is 512.
INFO mistralrs_core::gguf::content: Model config:
general.architecture: llama
general.base_model.0.name: Mistral Small 24B Base 2501
general.base_model.0.organization: Mistralai
general.base_model.0.repo_url: https://huggingface.co/mistralai/Mistral-Small-24B-Base-2501
general.base_model.0.version: 2501
general.base_model.count: 1
general.basename: Mistral-Small
general.file_type: 15
general.finetune: Instruct
general.languages: en, fr, de, es, it, pt, zh, ja, ru, ko
general.license: apache-2.0
general.name: Mistral Small 24B Instruct 2501
general.quantization_version: 2
general.size_label: 24B
general.type: model
general.version: 2501
llama.attention.head_count: 32
llama.attention.head_count_kv: 8
llama.attention.key_length: 128
llama.attention.layer_norm_rms_epsilon: 0.00001
llama.attention.value_length: 128
llama.block_count: 40
llama.context_length: 32768
llama.embedding_length: 5120
llama.feed_forward_length: 32768
llama.rope.dimension_count: 128
llama.rope.freq_base: 100000000
llama.vocab_size: 131072
quantize.imatrix.chunks_count: 128
quantize.imatrix.dataset: /training_dir/calibration_datav3.txt
quantize.imatrix.entries_count: 280
quantize.imatrix.file: /models_out/Mistral-Small-24B-Instruct-2501-GGUF/Mistral-Small-24B-Instruct-2501.imatrix
INFO mistralrs_core::utils::normal: DType selected is F16.
INFO mistralrs_core::pipeline::loaders: Using automatic device mapping parameters: text[max_seq_len: 4096, max_batch_size: 1].
INFO mistralrs_core::utils::log: Model has 40 repeating layers.
INFO mistralrs_core::utils::log: Loading model according to the following repeating layer mappings:
INFO mistralrs_core::utils::log: Layers 0-39: cpu
WARN mistralrs_core::pipeline::gguf: Device mapping contains a mix of GPU and CPU. There is no CPU support for PagedAttention, disabling PagedAttention.
INFO mistralrs_core::gguf::gguf_tokenizer: GGUF tokenizer model is `gpt2`, kind: `Bpe`, num tokens: 131072, num added tokens: 0, num merges: 269443, num scores: 0
INFO mistralrs_core::gguf::chat_template: Discovered and using GGUF chat template: `{{ bos_token }}{% for message in messages %}{% if message['role'] == 'user' %}{{ '[INST]' + message['content'] + '[/INST]' }}{% elif message['role'] == 'system' %}{{ '[SYSTEM_PROMPT]' + message['content'] + '[/SYSTEM_PROMPT]' }}{% elif message['role'] == 'assistant' %}{{ message['content'] + eos_token }}{% else %}{{ raise_exception('Only user, system and assistant roles are supported!') }}{% endif %}{% endfor %}`
INFO mistralrs_core::utils::normal: DType selected is F16.
INFO mistralrs_core::pipeline::paths: Using literal chat template.
INFO mistralrs_core::pipeline::chat_template: bos_toks = "<s>", eos_toks = "</s>", unk_tok = <unk>
INFO mistralrs_server: Model loaded.
INFO mistralrs_core: Beginning dummy run.
ERROR mistralrs_core::engine: prompt step - Model failed with error: Msg("shape mismatch on dim 3, 160 <> 128")
INFO mistralrs_core: Dummy run completed in 0.098900506s.
INFO mistralrs_server::interactive_mode: Starting interactive loop with sampling params: SamplingParams { temperature: Some(0.1), top_k: Some(32), top_p: Some(0.1), min_p: Some(0.05), top_n_logprobs: 0, frequency_penalty: Some(0.1), presence_penalty: Some(0.1), stop_toks: None, max_len: Some(4096), logits_bias: None, n_choices: 1, dry_params: Some(DrySamplingParams { sequence_breakers: ["\n", ":", "\"", "*"], multiplier: 0.0, base: 1.75, allowed_length: 2 }) }
====================
Welcome to interactive mode! Because this model is a text model, you can enter prompts and chat with the model.

Commands:
- `\help`: Display this message.
- `\exit`: Quit interactive mode.
- `\system <system message here>`:
    Add a system message to the chat without running the model.
    Ex: `\system Always respond as a pirate.`
====================
> Why is the sky blue?
ERROR mistralrs_core::engine: prompt step - Model failed with error: Msg("shape mismatch on dim 3, 160 <> 128")
ERROR mistralrs_server::interactive_mode: Got a model error: "shape mismatch on dim 3, 160 <> 128", response: ChatCompletionResponse { id: "1", choices: [Choice { finish_reason: "error", index: 0, message: ResponseMessage { content: Some(""), role: "assistant", tool_calls: [] }, logprobs: None }], created: 1738429138, model: "/download/location/Mistral-Small-24B-Instruct-2501-GGUF", system_fingerprint: "local", object: "chat.completion", usage: Usage { completion_tokens: 0, prompt_tokens: 17, total_tokens: 17, avg_tok_per_sec: 261.53848, avg_prompt_tok_per_sec: inf, avg_compl_tok_per_sec: NaN, total_time_sec: 0.065, total_prompt_time_sec: 0.0, total_completion_time_sec: 0.0 } }
```

## Latest commit or version
Latest, commit a8f83a01aa9102aae909972363124f4bb3a859ca

**Comment by EricLBuehler** [comment]
@Remember20240719 this should be fixed in #1131!

**Comment by Remember20240719** [comment]
@EricLBuehler it is fixed, thank you!

**Comment by EricLBuehler** [comment]
Glad to help - let me know if there are any other issues!
